platform/darwin/test/MGLSymbolStyleLayerTests.mm

```diff
@@ -1892,6 +1961,50 @@ - (void)testProperties {
                               @"textTransform should return the default value after being unset.");
     }
 
+    // text-variable-anchor
+    {
+        XCTAssertTrue(rawLayer->getTextVariableAnchor().isUndefined(),
+                      @"text-variable-anchor should be unset initially.");
+        NSExpression *defaultExpression = layer.textVariableAnchor;
+
+        NSExpression *constantExpression = [NSExpression expressionWithFormat:@"{'top','bottom'}"];
+        layer.textVariableAnchor = constantExpression;
+        mbgl::style::PropertyValue<std::vector<mbgl::style::SymbolAnchorType>> propertyValue = { { mbgl::style::SymbolAnchorType::Top, mbgl::style::SymbolAnchorType::Bottom } };
+        XCTAssertEqual(rawLayer->getTextVariableAnchor(), propertyValue,
+                       @"Setting textVariableAnchor to a constant value expression should update text-variable-anchor.");
+        XCTAssertEqualObjects(layer.textVariableAnchor, constantExpression,
+                              @"textVariableAnchor should round-trip constant value expressions.");
+
+        constantExpression = [NSExpression expressionWithFormat:@"{'top','bottom'}"];
```

Comment:
We're not really testing the Obj-C types here. Can we evaluate the MGL types?

Reply:
Similar pattern is applied to represent enum expressions across all these tests, e.g. https://github.com/mapbox/mapbox-gl-native/blob/master/platform/darwin/test/MGLSymbolStyleLayerTests.mm#L103